ProjFS.Mac/PrjFSKext/PrjFSKext/KauthHandler.cpp

```diff
@@ -539,6 +542,56 @@ static bool ShouldHandleEvent(
     return true;
 }
 
+static bool ShouldHandleFileOpEvent(
```

Comment:
It just felt like a lot of duplication, though I agree that breaking it down into a bunch of small helpers is probably overkill.

It appears like the vnode function is a superset of the fileop function. Could this just be broken into two methods, one of which calls the other? What I'd like to avoid is duplicated efforts in bug fixing, adding more common checks, etc over time.

Reply:
> It appears like the vnode function is a superset of the fileop function. Could this just be broken into two methods, one of which calls the other? 

I think we could have `ShouldHandleFileOpEvent` call `ShouldHandleVnodeOpEvent` and carefully pass in dummy values for `action` to avoid having it go down paths we don't want it to.  

There are some downsides to that approach though:

- kext would be performing extra work in a performance critical path
- Potentially more fragile as changes to `ShouldHandleVnodeOpEvent` could inadvertently break `ShouldHandleFileOpEvent`.

For long term maintainability I like the common helper methods approach as it helps keep clear exactly how we're making the "ShouldHandle" decision for each type of event.

> What I'd like to avoid is duplicated efforts in bug fixing, adding more common checks, etc over time.

I completely agree on this goal.  What would you think if we continue to monitor these functions as we implement the missing callbacks\notifications, and revisit this before considering #152 complete?

As I've started working on the other notifications (in parallel with this PR) I can already see how `ShouldHandleFileOpEvent` will need additional changes that might actually break the subset\superset relationship with `ShouldHandleVnodeOpEvent`.

If it's OK with you I think it would be cleaner to iterate on these functions separately and then decide the best way to share code once we have the basic set of notifications working.

Reply:
That makes sense to me :+1: